Classes/Controller/FrontEndEditorController.php

```diff
@@ -41,4 +49,33 @@ private function getUidOfLoggedInUser(): int
     {
         return $this->context->getPropertyFromAspect('frontend.user', 'id');
     }
+
+    public function editAction(Tea $tea): ResponseInterface
+    {
+        $this->checkIfUserIsOwner($tea);
+
+        $this->view->assign('tea', $tea);
+
+        return $this->htmlResponse();
+    }
+
+    /**
+     * @throws \RuntimeException
+     */
+    private function checkIfUserIsOwner(Tea $tea): void
+    {
+        if ($tea->getOwnerUid() !== $this->getUidOfLoggedInUser()) {
+            throw new \RuntimeException('You do not have the permissions to edit this tea.', 1687363749);
+        }
+    }
+
+    public function updateAction(Tea $tea): ResponseInterface
+    {
+        $this->checkIfUserIsOwner($tea);
+
+        $this->teaRepository->update($tea);
+        $this->persistenceManager->persistAll();
```

Comment:
Do we need the `persistAll()`? Shouldn't that be done by Extbase under the hood after the request was handled?
We do a redirect which (at least in earlier versions) finishes the request, in contrast to a forward.

Reply:
Indeed (at least in my V11 tests, am I'm not aware of any changes in this area). Done.